R/simulation-results.R

```diff
@@ -39,10 +39,15 @@ SimulationResults <- R6::R6Class("SimulationResults",
       rClr::clrCall(self$ref, "HasResultsFor", as.integer(individualId))
     },
 
-    getValuesByPath = function(path, individualIds) {
+    getValuesByPath = function(path, individualIds, stopIfNotFound = TRUE) {
       validateIsNumeric(individualIds)
       individualIds <- c(individualIds)
       values <- rClr::clrCall(self$ref, "AllValuesFor", path, as.integer(individualIds))
+
+      if (unique(is.nan(values)) && stopIfNotFound) {

```

Comment:
Assuming that rClr call returns NaNs if the results is not present.

Reply:
Nope. Check the tests. It returns NA

Reply:
No, it returns NaN and is converted to NA in the next step. The check for `stopIfNotFound` occurs before that.

Reply:
oh I did not see that this check was is IN the object itself. Is there a reason why you don't do the check in the calling method (e.g. the utilities?)

Reply:
Are we going to have one method for getting simulation results, or two - `getOutputValues` and `getOutputValuesTLF`? If it will be only one, I will put the check into one of them. Otherwise having the check here avoids duplication of code.

Reply:
Can't answer that yet. Ideally only one but  who knows